evcouplings/mutate/calculations.py

```diff
@@ -120,8 +126,50 @@ def _predict_mutant(mutation_str):
     else:
         mutations = pred.loc[:, mutant_column]
 
+    # if there is a segment column, use that to apply
+    # segment information to every mutation
+    if "segment" in pred.columns and pred.loc[:, "segment"].notnull().all():
```

Comment:
That was intentional, if the segment column is present but entirely empty we don't have any segment information for any of the mutations, so we just act as though the segment info is absent. This is necessary because single_mutant_matrix now always returns a df that has a column for segments, and this column is empty for monomers.

Reply:
ok